File: 2576/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(maxNumOfMarkeIndices([]int{1, 2, 3, 4}))
}

func maxNumOfMarkeIndices(nums []int) int {
	sort.Ints(nums)
	ans := 0 
	n := len(nums)
	m := n / 2
	for i, j := 0, m; i < m && j < n; i++ {
		for ; j < n && 2 * nums[i] > nums[j]; {
			j++
		}
		if j < n {
			ans += 2
			j++
		}
	}
	return ans
}

File: 2576/main.rs
struct Solution {}

impl Solution {
    pub fn max_num_of_marked_indices(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut ans = 0;
        let n = nums.len();
        let m = n / 2;
        let mut i = 0;
        let mut j = m;
        while i < m && j < n {
            while j < n && 2 * nums[i] > nums[j] {
                j += 1;
            }
            if j < n {
                ans += 2;
                j += 1;
            }
            i += 1;
        }
        return ans;
    }
}

fn main() {
    println!("{}", Solution::max_num_of_marke_indices(vec![1,2,3,4]));
}